enterprise/security/src/main/java/org/slf4j/impl/Neo4jLoggerFactory.java

```diff
@@ -0,0 +1,320 @@
+/*
+ * Copyright (c) 2002-2016 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.slf4j.impl;
+
+import org.slf4j.ILoggerFactory;
+import org.slf4j.Logger;
+import org.slf4j.helpers.MarkerIgnoringBase;
+import org.slf4j.helpers.NOPLogger;
+
+import java.util.concurrent.atomic.AtomicReference;
+
+import org.neo4j.logging.Log;
+import org.neo4j.logging.LogProvider;
+
+public class Neo4jLoggerFactory implements ILoggerFactory
+{
+    private final AtomicReference<LogProvider> logProvider = new AtomicReference<>();
+
+    @Override
+    public Logger getLogger( String name )
+    {
+        if ( logProvider.get() != null )
+        {
+            Log log = logProvider.get().getLog( name );
+            return new Neo4jLogger( name, log );
+        }
+        return NOPLogger.NOP_LOGGER;
+    }
+
+    public synchronized void setNeo4jLogProvider( LogProvider logProvider )
+    {
+        this.logProvider.set( logProvider );
+    }
+
+    private class Neo4jLogger extends MarkerIgnoringBase
+    {
+        private final String name;
+        private final Log log;
+
+        public Neo4jLogger( String name, Log log )
+        {
+            this.name = name;
+            this.log = log;
+        }
+
+        @Override
+        public String getName()
+        {
+            return name;
+        }
+
+        @Override
+        public boolean isTraceEnabled()
+        {
+            return false;
+        }
+
+        @Override
+        public void trace( String msg )
+        {
+        }
+
+        @Override
+        public void trace( String format, Object arg )
+        {
+        }
+
+        @Override
+        public void trace( String format, Object arg1, Object arg2 )
+        {
+        }
+
+        @Override
+        public void trace( String format, Object[] argArray )
+        {
+        }
+
+        @Override
+        public void trace( String msg, Throwable t )
```

Comment:
Are the `trace()` calls unimplemented intentionally? Perhaps they should throw `UnsupportedOperationException`?